stdpopsim/slim_engine.py

```diff
@@ -948,17 +948,58 @@ def fix_time(event):
     condition_on_allele_frequency = []
     op_id = stdpopsim.ext.ConditionOnAlleleFrequency.op_id
     slim_mutation_ids = []
-    for x in _dfe_to_mtypes(contig).values():
+    slim_mutation_dfe = []
+    for i, x in _dfe_to_mtypes(contig).items():
         slim_mutation_ids.extend([u[0] for u in x])
+        slim_mutation_dfe.extend([i for u in x])
     for ee in extended_events:
-        if hasattr(ee, "mutation_type_id"):
-            mt_id = getattr(ee, "mutation_type_id")
+        if hasattr(ee, "single_site_id"):
+            mut_id = getattr(ee, "single_site_id")
             cls_name = ee.__class__.__name__
-            if mt_id not in slim_mutation_ids:
```

Comment:
Wouldn't it be cleaner to check if there's a unique DFE with this ID, and then to check if that DFE has a single mutation type?